src/mailadm/bot.py

```diff
@@ -36,16 +36,41 @@ def __init__(self, db: DB, account: deltachat.Account):
         self.account = account
         with self.db.read_connection() as conn:
             config = conn.config
-            self.admingrpid = config.admingrpid
-            self.admingroup = account.get_chat_by_id(int(self.admingrpid))
+            self.admingrpid = int(config.admingrpid)
+            self.admingroup = account.get_chat_by_id(self.admingrpid)
 
     @account_hookimpl
     def ac_incoming_message(self, message: deltachat.Message):
         arguments = message.text.split(" ")
-        print("process_incoming command:", arguments)
-        if not self.check_privileges(message):
-            message.create_chat()
-            message.chat.send_text("Sorry, I only take commands from the admin group.")
+        print("process_incoming message:", message.text)
+        if not self.is_admin_group_message(message):
+            chat = message.create_chat()
+            if chat.is_group():
+                if message.get_sender_contact() not in self.admingroup.get_contacts():
+                    chat.send_text("Sorry, you can not contact me in a group chat. Please use a 1:1 chat.")
+                    chat.remove_contact(self.account.get_self_contact())   # leave group
+                elif message.quote:
+                    if message.quote.get_sender_contact().addr == self.account.get_config("addr"):
+                        recipient = chat.get_name()
+                        chat = self.account.create_chat(recipient)
+                        message.set_override_sender_name(self.account.get_config("addr"))
+                        chat.send_msg(message)
+                # else ignore message, it's just admins discussing what to do with the request
+            elif message.text[0] == "/":
+                chat.send_text("Sorry, I only take commands from the admin group.")
+            else:
+                name = message.get_sender_contact().addr
+                admins = self.admingroup.get_contacts()
+                admins.remove(self.account.get_self_contact())
+                supportgroup = None
+                for chat in self.account.get_chats():
+                    if chat.get_name() == name:
+                        if chat.is_group():
+                            supportgroup = chat
+                if not supportgroup:
+                    supportgroup = self.account.create_group_chat(name, admins, True)
+                message.set_override_sender_name(self.account.get_config("addr"))
+                supportgroup.send_msg(message)
```

Comment:
so a user that sent a question message, will result in the message being forwarded to the support group in the name of the bot?  Shouldn't it rather be a message where the override name is the original recipient so that support-group members can recognize where this is coming from?

Reply:
They can see the user's name in the group name - when the admbot creates the support group, it sets the group name to the user name.

I didn't really think of using the supportuser's name for the override, I though I had to use the bot's address for this, maybe I misunderstood how overrides work